cluster-autoscaler/simulator/predicates.go

```diff
@@ -282,21 +264,32 @@ func (pe *PredicateError) PredicateName() string {
 }
 
 // CheckPredicates checks if the given pod can be placed on the given node.
-func (p *PredicateChecker) CheckPredicates(pod *apiv1.Pod, nodeInfo *schedulernodeinfo.NodeInfo) *PredicateError {
-	for _, predInfo := range p.predicates {
-		// Skip affinity predicate if it has been disabled.
-		if !p.enableAffinityPredicate && predInfo.Name == affinityPredicateName {
-			continue
+func (p *PredicateChecker) CheckPredicates(pod *apiv1.Pod, nodeInfo *scheduler_nodeinfo.NodeInfo) *PredicateError {
+	state := scheduler_framework.NewCycleState()
```

Comment:
This doesn't actually contain the node we're considering (since it's not part of snapshot). Doesn't that basically negate any benefits of running PreFilters?

Reply:
Fixed in followup commit.